iceoryx2-ffi/c/src/api/service_builder_blackboard.rs

```diff
@@ -0,0 +1,912 @@
+
+            let service_builder = ManuallyDrop::into_inner(service_builder.blackboard_opener);
+            service_builder_struct.set(ServiceBuilderUnion::new_local_blackboard_opener(
+                service_builder.max_nodes(value),
+            ));
+        }
+    }
+}
+
+/// Adds key-value pairs to the blackboard.
+///
+/// # Arguments
+///
+/// * `service_builder_handle` - Must be a valid [`iox2_service_builder_blackboard_creator_h_ref`]
+///   obtained by
+///   [`iox2_service_builder_blackboard_creator`](crate::iox2_service_builder_blackboard_creator).
+/// * `key` - The key that shall be added to the blackboard
+/// * `value_ptr` - a valid, non-null [`*mut c_void`] pointer which points to the value to be stored
+/// * `release_callback` - a valid callback with [`iox2_service_blackboard_creator_add_release_callback`] signature for the `value_ptr` cleanup
+/// * `type_name` - the name of the value type
+/// * `type_name_len` - the length of `type_name`
+/// * `type_size` - the size of the value type to be added to the blackboard
+/// * `type_alignment` - the alignment of the value type to be added to the blackboard
+///
+/// # Safety
+///
+/// * `service_builder_handle` must be a valid and non-null handle
+/// * `value_ptr` is valid and non-null
+/// * `release_callback` must take care of the cleanup for `value_ptr`
+/// * `type_name`, `type_name_len`, `type_size` and `type_align` must satisfy the type details of the value type
+#[no_mangle]
+pub unsafe extern "C" fn iox2_service_builder_blackboard_creator_add(
+    service_builder_handle: iox2_service_builder_blackboard_creator_h_ref,
+    key: KeyFfi,
+    value_ptr: *mut c_void,
+    release_callback: iox2_service_blackboard_creator_add_release_callback,
+    type_name: *const c_char,
+    type_name_len: usize,
+    type_size: usize,
+    type_align: usize,
+) {
+    service_builder_handle.assert_non_null();
+    debug_assert!(!value_ptr.is_null());
+
+    let mut type_details = TypeDetail::new::<()>(TypeVariant::FixedSize);
+    iceoryx2::testing::type_detail_set_name(
+        &mut type_details,
+        TypeNameString::from_bytes_truncated(core::slice::from_raw_parts(
+            type_name.cast(),
+            type_name_len,
+        )),
+    );
+    iceoryx2::testing::type_detail_set_size(&mut type_details, type_size);
+    iceoryx2::testing::type_detail_set_alignment(&mut type_details, type_align);
+
+    let value_writer = Box::new(move |raw_memory_ptr: *mut u8| {
+        let ptrs = __internal_calculate_atomic_mgmt_and_payload_ptr(raw_memory_ptr, type_align);
+        core::ptr::copy_nonoverlapping(
+            value_ptr,
+            ptrs.atomic_payload_ptr as *mut c_void,
+            type_size,
+        );
+    });
+    let value_size = 2 * (align(type_size, type_align))
+        + align(core::mem::size_of::<UnrestrictedAtomicMgmt>(), type_align);
+    let value_alignment = max(core::mem::align_of::<UnrestrictedAtomicMgmt>(), type_align);
```

Comment:
Same here, move this into an internal function in the unrestricted management and test the function again. It shall return the same alignment as the `UnrestrictedAtomic<T>` when the corresponding type_align stuff is set.